wp-sw-cache/lib/service-worker.js

```diff
@@ -1,103 +1,127 @@
-(function(self){
-  var CACHE_PREFIX = '__wp-sw-cache::';
+(function(self, localforage){
 
-  var CACHE_NAME = CACHE_PREFIX + '$name';
-  var CACHE_FILES = $files;
+  // Represents the name of both storage and serviceworker
+  // This shouldn't change because the SW code below is smart enough to
+  // update what's needed and delete what isn't
+  var storageKey = '$name';
 
-  var debug = $debug;
+  // Outputs messages to the JS console if desired
+  var log = function() {
+    if($debug) {
+      console.log.apply(console, arguments);
+    }
+  };
 
-  self.addEventListener('install', function(event) {
-    // Perform install step:  loading each required file into cache
-    event.waitUntil(
-      caches.open(CACHE_NAME)
-        .then(function(cache) {
-          // Add all offline dependencies to the cache
-          if (debug) {
-            console.log('[install] Caches opened, adding all core components to cache');
-          }
-          return cache.addAll(CACHE_FILES);
-        })
-        .then(function() {
-          if (debug) {
-            console.log('[install] All required resources have been cached, we\'re good!');
-          }
-          return self.skipWaiting();
-        })
-    );
-  });
+  var wpSwCache = {
+    // Hash representing URLs and their modified times
+    // A changed hash will ensure the SW is downloaded when URLs change
+    version: '$version',
```

Comment:
Actually, a change in your `urls` (the `hash` part) property will trigger the update. The version should act as a guard to prevent your install / activate handlers from running when other parts of the sw have changed but not yours.